src/teaching-element/Assessment/DragDrop.vue

```diff
@@ -126,6 +128,25 @@ export default {
       this.userAnswer[id] = this.userAnswer[id].filter(a => a.id !== answer.id);
       this.answersCollection.push(answer);
     },
+    retakeIncorrect() {
+      Object.keys(this.userAnswer).forEach(groupId => {
+        let wrongAnswers = [];
+        this.userAnswer[groupId].forEach((answer) => {
+          if (!includes(this.correct[groupId], answer.id)) {
+            wrongAnswers.push(answer.id);
+            this.answersCollection.push(answer);
+          }
+        });
+        this.removeFromGroup(wrongAnswers, groupId);
```

Comment:
If `wrongAnswers` are only used to remove from the `userAnswers` prop then can't the `answersCollection` prop be used instead to remove them?